pkg/clients/atlan/client.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright 2025.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package atlan
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"slices"
+	"time"
+
+	atlansdk "github.com/atlanhq/atlan-go/atlan/assets"
+	"github.com/gojek/heimdall/v7"
+	"github.com/redhat-data-and-ai/usernaut/pkg/request"
+	"github.com/redhat-data-and-ai/usernaut/pkg/request/httpclient"
+	"github.com/redhat-data-and-ai/usernaut/pkg/utils"
+)
+
+// NewClient creates a new Atlan client with simple API token authentication
+func NewClient(atlanAppConfig map[string]interface{},
+	connectionPoolConfig httpclient.ConnectionPoolConfig,
+	hystrixResiliencyConfig httpclient.HystrixResiliencyConfig) (*AtlanClient, error) {
+
+	atlanConfig := AtlanConfig{}
+	if err := utils.MapToStruct(atlanAppConfig, &atlanConfig); err != nil {
+		return nil, fmt.Errorf("failed to parse atlan configuration: %w", err)
+	}
+
+	// Validate required fields
+	if atlanConfig.URL == "" {
+		return nil, fmt.Errorf("atlan configuration is missing required field: URL")
+	}
+	if atlanConfig.APIToken == "" {
+		return nil, fmt.Errorf("atlan configuration is missing required field: APIToken")
+	}
+
+	// Initialize HTTP client without certificates (Atlan uses API token, not certs)
+	client, err := httpclient.InitializeClient(
+		"atlan",
+		connectionPoolConfig,
+		hystrixResiliencyConfig,
+		heimdall.NewRetrier(heimdall.NewConstantBackoff(100*time.Millisecond, 50*time.Millisecond)), // retry logic
+		3,
+		nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize http client: %w", err)
+	}
+
+	// Initialize the Atlan SDK client for operations that require it (e.g., DeleteUser)
+	sdkClient, err := atlansdk.Context(atlanConfig.URL, atlanConfig.APIToken)
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize Atlan SDK client: %w", err)
+	}
+
+	return &AtlanClient{
+		client:                client,
+		sdkClient:             sdkClient,
+		url:                   atlanConfig.URL,
+		apiToken:              atlanConfig.APIToken,
+		identityProviderAlias: atlanConfig.IdentityProviderAlias,
```

Comment:
before creating the client, do we want to do a quick check on the `identityProviderAlias` variable with it's possible values?